src/main/java/org/example/chess/piece/Horse.java

```diff
@@ -0,0 +1,64 @@
+package org.example.chess.piece;
+
+import org.example.chess.Field;
+import org.example.chess.piece.util.ChessPieceFieldValidator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+final class Horse extends ChessPiece{
+    Horse(Color color, Field currentPosition) {
+        super(color, currentPosition);
+    }
+
+    @Override
+    Set<Field> possibleMoves() {
+        Set<Field> fieldsOfPossibleMoves = new HashSet<>();
+
+        checkMoves(fieldsOfPossibleMoves, "up");
+        checkMoves(fieldsOfPossibleMoves, "down");
+        checkMoves(fieldsOfPossibleMoves, "right");
+        checkMoves(fieldsOfPossibleMoves, "left");
+
+        return fieldsOfPossibleMoves;
+    }
+
+
+    private static Set<Field> checkMoves(Set<Field> fields, String command) {
+        int cycles = 2;
+        int row = 5;
+        int col = 3;
+        if (command.equals("up") || command.equals("down")) {
+            int direction = command.equals("up") ? 1 : -1;
+
+            while (cycles > 0) {
+                cycles--;
+                row += direction;
+            }
+
+            boolean isLeftSideEmpty = ChessPieceFieldValidator.isValid(row, col - 1)
+                    ? fields.add(new Field(row, col + 1))
```

Comment:
This way of adding possible moves is a little bit obfuscated, in my view it would be better if the computed valid moves were returned from this method

Reply:
yes it could be easily changed, i just wanted to use that functionality, the most.

Reply:
The functionallity - you mean ternary operator? It doesn't seem a good fit for doing side effects, in my view at least